src/simplify_algebra.cpp

```diff
@@ -1066,10 +1066,16 @@ struct find_split_reshape
         }
 
         std::vector<instruction_ref> vec_rsp(split_outputs.size());
-        std::transform(split_outputs.begin(), split_outputs.end(), vec_rsp.begin(), [](auto i) {
+        std::list<instruction> undef_inss{instruction(make_op("undefined"), {}, {})};
```

Comment:
This is the wrong way to insert undefined operator(as the instruction doesnt belong to any module in the program. Why are you inserting undefined operators in the first place? Why dont we just exit when output size is not 1?

Reply:
The check for output size happens in the lambda function for each item in split_outputs and so adding the undefined instruction is to ensure that the following `same_ops` check fails causing it to exit

Reply:
I think it would be better to do `std::any_of` and check for the output size and then exit.